dev/docs/windows-source-installation.md

```diff
@@ -0,0 +1,81 @@
+
+_**Speed Tip:**_ If the complete repository history isn't needed then using a shallow clone (`git clone --depth 1`) will
+save significant time.
+
+[TODO 1]: Get the spark, mvn and java download installation
+Apache SystemDS requires dependencies from the other Apache projects. To download these sources,
+run one of the `getSparkMvnJdk` scripts located in the `<SYSTEMDS_HOME>` directory. These scripts clone their respective
+*master* branches.
+* `getSpark.bat` for Windows.
+
+## Building SystemDS source code
+
+IntelliJ IDEA is preferred for best developer experience.
+
+### Opening the IntelliJ Source Code for Build
+
+Using IntelliJ IDEA **File | Open**, select the `<SYSTEMDS_HOME>` directory. 
+* If IntelliJ IDEA displays an error about a missing or out of date required plugin (e.g. maven),
+  [enable, upgrade, or install that plugin](https://www.jetbrains.com/help/idea/managing-plugins.html) and restart IntelliJ IDEA.
+
+### IntelliJ Build Configuration
+
+JDK version 1.8 (u151 or newer) is required for building and developing for SystemDS developement.
+
+1. Using IntelliJ IDEA, [configure](https://www.jetbrains.com/help/idea/sdk.html) a JDK named "**1.8**", pointing to `<JDK_18_HOME>`.
+   * If not already present, add `<JDK_18_HOME>/lib/tools.jar` [to the Classpath](https://www.jetbrains.com/help/idea/sdk.html#manage_sdks) tab
+     for the **1.8** JDK.
+2. If the _Maven Integration_ plugin is disabled, [add the path variable](https://www.jetbrains.com/help/idea/working-with-projects.html#path-variables)
+   "**MAVEN_REPOSITORY**" pointing to `<USER_HOME>/.m2/repository` directory.
+3. _**Speed Tip:**_ If you have enough RAM on your computer,
+   [configure the compiler settings](https://www.jetbrains.com/help/idea/specifying-compilation-settings.html)
+   to enable the "Compile independent modules in parallel" option. Also set the "User-local build process VM options" to `-Xmx2G`.
+   These changes will greatly reduce the compile time.
+4. Now, selecting the IntelliJ IDEA **Build | Build module 'systemds'** option starts the maven build.
+5. _**Speed Tip:**_ If the development machine have enough RAM, [configure the compiler settings](https://www.jetbrains.com/help/idea/specifying-compilation-settings.html)
+   to enable the "Compile independent modules in parallel" option.
+ 
+### Building the Source Code
+To build SystemDS from source, choose **Build | Build Project** from the main menu.
+OR
+To maven build, run the `mvn` command in `<SYSTEMDS_HOME>` directory. See the `pom.xml` file for details.
+
+## Testing
+To run the SystemDS built from source, choose **Run | Run** from the main menu.
+
+To run tests on the build, apply these setting to the **Run | Edit Configurations... | Defaults | JUnit** configuration tab:
+  * Working dir: `<SYSTEMDS_HOME>`
+  * VM options:
+    * `-ea`
+    * `-junit4`
+    * `-Djava.system.class.loader=com.intellij.rt.execution.junit.JUnitStarter`
```

Comment:
correct me if i'm wrong. But I'm not sure if this executes the tests like the current setup.
It seems to me this is directly using junit for the execution, if that is correct then I'm not sure if it efficiently parallelize the execution of the tests, like if executed with the mvn surefire setup we use in integration testing.

Reply:
Even I do have a doubt. Only the `-ea` option seems to be fine for me. I shall remove the two flags.